homeassistant/components/stream/__init__.py

```diff
@@ -50,11 +50,7 @@ def request_stream(hass, stream_source, *, fmt="hls", keepalive=False, options=N
         options = {}
 
     # For RTSP streams, prefer TCP
-    if (
-        isinstance(stream_source, str)
-        and stream_source[:7] == "rtsp://"
-        and not options
-    ):
+    if isinstance(stream_source, str) and stream_source[:7] == "rtsp://":
         options["rtsp_flags"] = "prefer_tcp"
         options["stimeout"] = "5000000"
```

Comment:
We shouldn't override any option that is passed in so only set if it is not set. We also should not assume that we're allowed to change the host dict. As with your current code for example, that is writing to the source. We should make a copy.

```suggestion
        options = {
            "rtsp_flags": "prefer_tcp",
            "stimeout": "5000000",
            **options,
        }
```